enhancements/ocp-coreos-layering/ocp-coreos-layering.md

```diff
@@ -0,0 +1,725 @@
+
+They use this as part of a `Dockerfile` build that overrides the kernel, and just as above in the 3rd party scanner case, they find the digested pull spec and override via a `MachineConfig` with custom `osImageURL`.
+
+Later, a fixed kernel is released in the official `registry.redhat.io/rhel-coreos-8:4.12.8` image. The customer (and support) can easily see the kernel version in that image, and know with confidence that they can upgrade to a default cluster level of `4.12.8`.
+Specifically, the customer uses `oc adm upgrade` (or the GUI), and after that is complete they remove the `MachineConfig` object, which removes all the OS level overrides.
+
+#### Externally built image
+
+As we move towards having users manage many clusters (10, 100 or more), it will make sense to support building a node image centrally. This will allow submitting the image to a security scanner or review by a security team before deploying to clusters.
+
+Acme Corp has 300 clusters distributed across their manufacturing centers. They want to centralize their build system in their main data center, and just distribute those images to single node edge machines. They provide a `custom-coreos-imagestream` object at installation time, and their node CoreOS image is deployed during the installation of each cluster without a build operation.
+
+In the future, we may additionally support a signature mechanism (IMA/fs-verity) that allows these images can be signed centrally.  Each node verifies this image rather than generating a custom state.
+
+(Note some unanswered questions below)
+
+### External image build workflow
+
+The administrator will see a number of documented example Dockerfiles, like https://github.com/coreos/coreos-layering-examples/
+
+They can copy and extend these and build them using any container build system they like.
+The administrator can also use non-Dockerfile build systems such as Tekton or buildah.
+
+The result will be a container image that can be passed via an API to clusters, in a similar way as way user workloads (pods).
+
+The build and test workflow could resemble something like this:
+<!--Diagram Source: https://docs.google.com/drawings/d/1OE1-h5SiABVGG2WtILTT99tG7FawcVhsFPVLNf58ufM/edit-->
+![External image build workflow](./customer_base_image_build.svg)
+
+### API Extensions ([MCO-280](https://issues.redhat.com/browse/MCO-280))
+
+Note we will continue to support MachineConfig.  The key goal is to allow the user to override the rhel-coreos base image.
+
+#### Current default proposal: Configure via imagestream
+
+This is currently prototyped in the https://github.com/openshift/machine-config-operator/tree/layering branch.  Here there’s an imagestream per MachineConfigPool, and the administrator can specify an image which overrides the default rhel-coreos base image.
+
+##### User workflow in layering model
```

Comment:
The next few sections cover image stream based flows. Maybe such sections should be explicitly moved to a future section? Since I know what the plan for phase 0 is, I can read or understand those sections but someone not so close to the team may have a hard time understanding what's for now and what's for later.

Reply:
done